### PR TITLE
feat: change the way we show amount and token type depending on metadata result

### PR DESCRIPTION
**Pull request by pedroferreira1**
Not NFT screens

image
image
image

NFT screens

image
image
image

### CODE REVIEW

File: src/screens/AddressDetail.js [hunk 1]
```diff
@@ -263,13 +270,23 @@ class AddressDetail extends React.Component {
     });
   }
 
+  getSelectedTokenMetadata = (selectedToken) => {
+    metadataApi.getDag(selectedToken).then((data) => {
```
Comment (msbrogli) [inline src/screens/AddressDetail.js:274]: `getDag`? I feel we should rename it to `getDagMetadata()`.

File: src/screens/AddressDetail.js [hunk 2]
```diff
@@ -263,13 +270,23 @@ class AddressDetail extends React.Component {
...
+      if (data) {
+        this.setState({ selectedTokenMetadata: data });
+      }
+    }).finally(() => {
+      this.setState({ metadataLoaded: true });
```
Comment (msbrogli) [inline src/screens/AddressDetail.js:279]: What if an error occurs? Should we still set `metadataLoaded` to true?

File: src/screens/TokenDetail.js [hunk 3]
```diff
@@ -94,6 +96,8 @@ class TokenDetail extends React.Component {
           }
         });
       }
+    }).finally(() => {
+      this.setState({ metadataLoaded: true });
```
Comment (msbrogli) [inline src/screens/TokenDetail.js:100]: Same thing here.

File: src/utils/helpers.js [hunk 4]
```diff
@@ -218,8 +218,25 @@ const helpers = {
...
+   * Render value to integer or decimal
+   *
+   * @param {number} amount Amount to render
+   * @param {boolean} isInteger If it's an integer or decimal
+   *
+   * @return {string} rendered value
+   * @memberof Helpers
+   * @inner
+   */
+  renderValue(amount, isInteger) {
```
Comment (msbrogli) [inline src/utils/helpers.js:233]: I think this method should be `renderValue(amount, decimals)`. But I understand that it just calls the library which does not support decimals yet, right?

**Review by msbrogli — DISMISSED** [review]

File: src/screens/AddressDetail.js (hunk 2 above)
Comment (pedroferreira1) [reply src/screens/AddressDetail.js:279]: If the metadata API returns 404 it is loaded as well. But maybe I should capture the error and consider it loaded only if it's 404

File: src/utils/helpers.js (hunk 4 above)
Comment (pedroferreira1) [reply src/utils/helpers.js:233]: Yes, exactly. Right now the lib supports only 2 decimals or 0 decimals. I guess if we add custom decimal places we will change that (and many other places). I think we should leave that for the future, what do you think?

File: src/screens/AddressDetail.js (hunk 1 above)
Comment (pedroferreira1) [reply src/screens/AddressDetail.js:274]: Makes sense. Changed that

File: src/screens/AddressDetail.js (hunk 2 above)
Comment (pedroferreira1) [reply src/screens/AddressDetail.js:278]: Done

File: src/screens/TokenDetail.js (hunk 3 above)
Comment (pedroferreira1) [reply src/screens/TokenDetail.js:99]: Done

File: src/utils/helpers.js (hunk 4 above)
Comment (msbrogli) [reply src/utils/helpers.js:233]: I agree.

**Review by andreabadesso — APPROVED** [review]

**Review by msbrogli — APPROVED** [review]

**Pull request merged**